src/main/java/de/tum/in/www1/artemis/service/iris/IrisGPT3_5Service.java

```diff
@@ -0,0 +1,202 @@
+package de.tum.in.www1.artemis.service.iris;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.tum.in.www1.artemis.domain.iris.IrisMessage;
+import de.tum.in.www1.artemis.domain.iris.IrisMessageContent;
+import de.tum.in.www1.artemis.domain.iris.IrisMessageSender;
+import de.tum.in.www1.artemis.domain.iris.IrisSession;
+
+@Service
+public class IrisGPT3_5Service implements IrisModel {
```

Comment:
I agree with Teamscale here. A JavaDoc comment explaining what this class does would probably make sense.
This would ideally contain some specifics about what sets GPT3.5 apart.